### PR TITLE
Multi DB Connection Fixes

### LINKED ISSUE #127: Custom database connection ignored.
**Describe the bug**
In trait `DatabaseTraits.php` the `__construct` method is never called:

```php
    public function __construct($attributes = [])
    {
        parent::__construct($attributes);
        if ($connection = config('roles.connection')) {
            $this->connection = $connection;
        }
    }
```
This is because `Role::class` & `Permission::class` Models extends abstract `Model::class` that overrides the traits `__construct`.

**Resolution**

1) Convert `DatabaseTrait` to `Database::class` that extends `Model::class`
2) `Role::class` & `Permission::class` extend `Database::class` instead of `Model::class`

### PR DESCRIPTION
**Pull request by marcnewton**
- Fixes #127
- Added Detection of user table name for foreign key constraints in migrations.

### CODE REVIEW

**Comment by jeremykenedy** [comment]
Thank you

File: src/Database/Migrations/2016_02_09_132439_create_permission_user_table.php [hunk 1]
```diff
@@ -17,14 +17,15 @@ public function up()
...
         $tableCheck = Schema::connection($connection)->hasTable($table);
+        $userTable = app(config('auth.providers.users.model'))->getTable();
 
         if (!$tableCheck) {
-            Schema::connection($connection)->create($table, function (Blueprint $table) use ($permissionsTable) {
+            Schema::connection($connection)->create($table, function (Blueprint $table) use ($permissionsTable, $userTable) {
                 $table->increments('id')->unsigned();
-                $table->integer('permission_id')->unsigned()->index();
-                $table->foreign('permission_id')->references('id')->on($permissionsTable)->onDelete('cascade');
+                $table->integer('role_id')->unsigned()->index();
```
Comment (imhmdb) [inline src/Database/Migrations/2016_02_09_132439_create_permission_user_table.php:25]: I don't get why this was changed, can you explain?

Comment (marcnewton) [reply src/Database/Migrations/2016_02_09_132439_create_permission_user_table.php:25]: This is indeed incorrect, it should be `permission_id` not `role_id`.
I checked my history, a mistake I made that was undone has been merged in the wrong order.

Comment (imhmdb) [reply src/Database/Migrations/2016_02_09_132439_create_permission_user_table.php:25]: how was this merged without proper reviewing 🤦🏽, a package for roles and permissions should insure stability for production usage.

Comment (jeremykenedy) [reply src/Database/Migrations/2016_02_09_132439_create_permission_user_table.php:25]: fixed.

**Pull request merged**